### ISSUE TITLE
Error when starting Tuleap in docker

### ISSUE THREAD

**Issue by RealSnowbird** [Enalean/docker-tuleap-aio#35]
Hi,

I've tried to start up Tuleap in Docker and faced the following issues:

* several error messages (e. g. "PHP Fatal error:  Call to undefined function random_bytes", "chown: invalid user: `mysql'", "egrep: /etc/tuleap/conf/database.inc: No such file or directory"); please see log below;
* Eventually a prompt appears, asking for entering a password. None of the passwords for MySQL or Coendiadm did the trick.

Is this a bug or is there something I can do?

I'm using Docker 17.03.1-ce on CentOs 7.

Thanks,
Norbert

```
[vmware@devdocker ~]$ sudo su
[sudo] password for vmware: 
[root@devdocker vmware]# docker volume create --name tuleap-data
tuleap-data
[root@devdocker vmware]# docker run -ti -e VIRTUAL_HOST=localhost -P -v tuleap-data:/data enalean/tuleap-aio
Unable to find image 'enalean/tuleap-aio:latest' locally
latest: Pulling from enalean/tuleap-aio
b26de5a391ad: Pull complete 
21dabc0355f9: Pull complete 
79caa8c91849: Pull complete 
13b40ecf9ecc: Pull complete 
dbc92eb11c79: Pull complete 
715f5c83bdd1: Pull complete 
Digest: sha256:22ac2beb22244f1df7a15e91bbcc09064a8ae79db5d00327093099324b14725b
Status: Downloaded newer image for enalean/tuleap-aio:latest
+ TULEAP_INSTALL_TIME=false
+ '[' '!' -f /data/etc/tuleap/conf/local.inc ']'
+ TULEAP_INSTALL_TIME=true
+ ./boot-install.sh
mv: inter-device move failed: `/var/lib/mysql' to `/data/lib/mysql'; unable to remove target: Directory not empty
mv: inter-device move failed: `/var/lib/gitolite' to `/data/lib/gitolite'; unable to remove target: Directory not empty
/bin/ls: cannot access /root/app/patches/*.patch: No such file or directory
Good! You are running CentOS 6.9

Configuration questions

Changing password for user codendiadm.
passwd: all authentication tokens updated successfully.
Installing Tuleap configuration files...
Renaming existing Apache configuration files...
Installing Apache configuration files...
Installing log files rotation...
Installing NSS configuration files...
Creating MySQL conf file...
Initializing MySQL: You can ignore additionnal messages on MySQL below this line:
***************************************
Stopping mysqld:                                           [  OK  ]
Initializing MySQL database:  Installing MySQL system tables...
/usr/libexec/mysqld: Got an error reading communication packets
OK
Filling help tables...
/usr/libexec/mysqld: Got an error reading communication packets
OK

To start mysqld at boot time you have to copy
support-files/mysql.server to the right place for your system

PLEASE REMEMBER TO SET A PASSWORD FOR THE MySQL root USER !
To do so, start the server, then issue the following commands:

/usr/bin/mysqladmin -u root password 'new-password'
/usr/bin/mysqladmin -u root -h b82d1a88bcb7 password 'new-password'

Alternatively you can run:
/usr/bin/mysql_secure_installation

which will also give you the option of removing the test
databases and anonymous user created by default.  This is
strongly recommended for production servers.

See the manual for more instructions.

You can start the MySQL daemon with:
cd /usr ; /usr/bin/mysqld_safe &

You can test the MySQL daemon with mysql-test-run.pl
cd /usr/mysql-test ; perl mysql-test-run.pl

Please report any problems with the /usr/bin/mysqlbug script!

                                                           [  OK  ]
Starting mysqld:                                           [  OK  ]
***************************************
Creating the Tuleap database...
Populating the Tuleap database...
PHP Fatal error:  Call to undefined function random_bytes() in /usr/share/tuleap/src/common/user/RandomNumberGenerator.class.php on line 38
Configuring the CVS server and CVS tracking tools...
/bin/cp: cannot stat `/etc/xinetd.d/cvs': No such file or directory
Stopping xinetd:                                           [FAILED]
Starting xinetd: /etc/init.d/xinetd: line 66: warning: setlocale: LC_TIME: cannot change locale (en_US): No such file or directory
/etc/init.d/xinetd: line 67: warning: setlocale: LC_ALL: cannot change locale (en_US): No such file or directory
/etc/init.d/xinetd: line 72: warning: setlocale: LC_ALL: cannot change locale (en_US): No such file or directory
/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US): No such file or directory
                                                           [  OK  ]
Updating daily cron job in system crontab...
Use same-host project web sites
Restore application cache directories:
info Recreating /var/tmp/tuleap_cache/lang
info Restore ownership to /var/tmp/tuleap_cache
info Restore ownership to /var/tmp/tuleap_cache/lang
info Restore ownership to /var/tmp/tuleap_cache/hooks.json
Invalidate application caches:
info Invalidate template engine cache
info Invalidate Restler cache
info Invalidate language cache
info Invalidate WSDL
info Invalidate Plugin hooks
Starting Tuleap: [OK]
Stopping httpd:                                            [  OK  ]
Starting httpd: Syntax error on line 65 of /etc/httpd/conf.d/tuleap-vhost.conf:
SSLCertificateFile: file '/etc/pki/tls/certs/localhost.crt' does not exist or is empty
                                                           [FAILED]
Starting crond:                                            [  OK  ]
Stopping nscd:                                             [FAILED]
Starting nscd:                                             [  OK  ]
nscd: 703 monitoring file `/etc/passwd` (1)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/group` (3)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/hosts` (4)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/resolv.conf` (5)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/services` (6)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 disabled inotify-based monitoring for file `/etc/netgroup': No such file or directory
nscd: 703 disabled monitoring for file `/etc/netgroup': No such file or directory
Install tuleap plugins
Install Docman
Install tracker
Import succeeded
Import succeeded
Import succeeded
error: The short name 'patch' is already used. Please use another one.
Import succeeded
Import succeeded
Import succeeded
Import succeeded
Import succeeded
Install Graphontrackersv5
Install AgileDashboard
Install Git
Creating Scrum template
INFO - [doRecordOnly] b201007220743_add_table_notification_delegation
INFO - [doRecordOnly] b201007291109_modify_pk_table_notification_delegation
INFO - [doRecordOnly] b201008200802_reduce_load_on_user_table
INFO - [doRecordOnly] b201009280731_add_table_groups_notif_delegation_message
INFO - [doRecordOnly] b201010191436_add_table_frs_file_deleted
INFO - [doRecordOnly] b201010201546_fill_frs_file_deleted
INFO - [doRecordOnly] b201011230835_add_column_format_to_artifact_history
INFO - [doRecordOnly] b201012140821_improve_frs_file
INFO - [doRecordOnly] b201012240808_add_table_frs_log
INFO - [doRecordOnly] b201103081738_add_column_filepath_to_frs_file
INFO - [doRecordOnly] b201103281122_add_wiki_attachment_delete_mechanism
INFO - [doRecordOnly] b201108241423_remove_slmbug_and_story
INFO - [doRecordOnly] b201108311456_add_image_renderer
INFO - [doRecordOnly] b201111021759_id_sharing
INFO - [doRecordOnly] b201201181909_add_index_on_svn_commits_table
INFO - [doRecordOnly] b201201231552_add_table_soap_limits
INFO - [doRecordOnly] b201203140933_add_notif_svn_per_branch
INFO - [doRecordOnly] b201207051342_add_has_avatar_column
INFO - [doRecordOnly] b201208241130_add_anonymous_in_filemodule_monitor
INFO - [doRecordOnly] b201209131359_add_ugroup_binding_column
INFO - [doRecordOnly] b201210121809_purge_bad_sshkeys_from_db
INFO - [doRecordOnly] b201302091402_add_frs_file_comment
INFO - [doRecordOnly] b201303121639_add_owner_to_system_events
INFO - [doRecordOnly] b201305071724_add_svn_change_log_option
INFO - [doRecordOnly] b201305161743_remove_unique_from_svn_checkins
INFO - [doRecordOnly] b201306110951_add_parent_child_project_link_table
INFO - [doRecordOnly] b201306171145_add_generic_user_table
INFO - [doRecordOnly] b201310171433_add_rest_authentication_token_table
INFO - [doRecordOnly] b201311081011_remove_codex_themes
INFO - [doRecordOnly] b201403061540_add_forge_permissions_table
INFO - [doRecordOnly] b201404281427_rename_experimental_to_flamingparrot
INFO - [doRecordOnly] b201405051118_add_svn_accessfile_history_table_with_first_version
INFO - [doRecordOnly] b201405051133_modify_groups_table_to_use_svn_accessfile_history_id
INFO - [doRecordOnly] b201405061202_modify_svn_accessfile_history_table
INFO - [doRecordOnly] b201405071410_replace_svn_accessfile_column_name_in_groups
INFO - [doRecordOnly] b201405071644_replace_version_date_column_values_by_timestamp
INFO - [doRecordOnly] b201405141602_remove_fontsize_preference
INFO - [doRecordOnly] b201406041516_add_email_gateway_salt
INFO - [doRecordOnly] b201407040902_old_user_do_not_see_welcome_tour
INFO - [doRecordOnly] b201407041153_add_tour_usage_statistic_table
INFO - [doRecordOnly] b201407081125_add_missing_column_in_tour_usage_statistics
INFO - [doRecordOnly] b201410301650_add_commit_to_tag_denied_in_groups
INFO - [doRecordOnly] b201503061743_add_homepage
INFO - [doRecordOnly] b201503061743_add_homepage_headline
INFO - [doRecordOnly] b201503271456_add_forgeconfig
INFO - [doRecordOnly] b201504081720_allow_restricted_project_access
INFO - [doRecordOnly] b201504091644_add_authenticated_user_group
INFO - [doRecordOnly] b201504171343_restore_FRS_owner_group
INFO - [doRecordOnly] b201505051630_add_new_password_field
INFO - [doRecordOnly] b201505131103_project_admin_can_choose_visibility
INFO - [doRecordOnly] b201505302300_remove_twitter_widgets
INFO - [doRecordOnly] b201506011450_increase_unix_password_field_size
INFO - [doRecordOnly] b201506031610_add_svn_immutable_tags_whitelist_table
INFO - [doRecordOnly] b201506031658_enlarge_your_tinyint
INFO - [doRecordOnly] b201507152200_increase_session_ip_field_size
INFO - [doRecordOnly] b201508141600_remove_people_skills
INFO - [doRecordOnly] b201508171048_add_svn_immutable_tags_table
INFO - [doRecordOnly] b201508241048_add_truncated_emails_column
INFO - [doRecordOnly] b201508271443_add_index_on_system_event_table
INFO - [doRecordOnly] b201509020939_drop_priority_plugin_hook
INFO - [doRecordOnly] b201510011056_add_index_on_service_table
INFO - [doRecordOnly] b201510131542_turn_user_groups_tables_innodb
INFO - [doRecordOnly] b201511231728_add_trovecat_mandatory_column
INFO - [doRecordOnly] b201512171700_add_svn_token_table
INFO - [doRecordOnly] b201512211530_force_refresh_svnaccessfile
INFO - [doRecordOnly] b201512281119_add_svn_token_usage_table
INFO - [doRecordOnly] b201512291430_add_indexes_for_statistics
INFO - [doRecordOnly] b201601051638_add_trovecat_display_column
INFO - [doRecordOnly] b201607221530_cleanup_csrf_token_from_userprefs
INFO - [doRecordOnly] b201607281135_create_plugin_frs_global_permissions
INFO - [doRecordOnly] b201607281200_migrate_existing_permissions
INFO - [doRecordOnly] b201608041530_migrate_existing_read_permissions
INFO - [doRecordOnly] b201608091610_add_index_for_frs_permissions
INFO - [doRecordOnly] b201608291800_migrate_incoherent_permissions_for_release
INFO - [doRecordOnly] b201608298050_migrate_incoherent_permissions
INFO - [doRecordOnly] b201609082200_remove_admssw_plugin
INFO - [doRecordOnly] b201609082230_remove_boomerang_plugin
INFO - [doRecordOnly] b201609082230_remove_foafprofiles_plugin
INFO - [doRecordOnly] b201609131045_update_session_management
INFO - [doRecordOnly] b201609221505_add_missing_permissions_french_wiki_pages
INFO - [doRecordOnly] b201610121840_remove_fusionforgecompat_plugin
INFO - [doRecordOnly] b201610241145_update_frs_admin_description
INFO - [doRecordOnly] b201611031112_add_svn_cache_parameter_table
INFO - [doRecordOnly] b201612231324_update_trove_categories_incorrect_values
INFO - [doRecordOnly] b201701131030_create_table_user_lost_password
INFO - [doRecordOnly] b201701231415_update_public_forum_in_private_projects
INFO - [doRecordOnly] b201701251415_user_lost_password_add_creation_date
INFO - [doRecordOnly] b201701311615_feedback_supports_new_session_identifier
INFO - [doRecordOnly] b20170207_remove_snippet_reference
INFO - [doRecordOnly] b201703131130_create_table_project_webhook_url
INFO - [doRecordOnly] b201703141400_create_table_project_webhook_log
INFO - [doRecordOnly] b201703201030_add_project_webhook_name
INFO - [doRecordOnly] b201703281110_keep_only_modperl_svn_authentication_mode
INFO - [doRecordOnly] b201704041400_clean_recent_element_user_preferences
INFO - [doRecordOnly] b201704051330_remove_binary_serialized_feedback
INFO - [doRecordOnly] b201705031528_create_table_user_dashboards
INFO - [doRecordOnly] b201705110903_create_table_project_dashboards
INFO - [doRecordOnly] b201705121210_update_table_user_dashboards
INFO - [doRecordOnly] b201705121719_update_table_project_dashboards
INFO - [doRecordOnly] b201705161407_create_tables_dashboards_lines
INFO - [doRecordOnly] b201705161407_create_tables_dashboards_lines_columns
INFO - [doRecordOnly] b201705161407_create_tables_dashboards_lines_columns_widgets
INFO - [doRecordOnly] b201705181212_store_widget_name
INFO - [doRecordOnly] b201705231141_store_widget_is_minimized
INFO - [doRecordOnly] b201706091011_migrate_old_dashboard
INFO - [doRecordOnly] b201706301720_remove_docman_v1
INFO - [doRecordOnly] b201708180900_remove_invalid_china_timezones
INFO - [doRecordOnly] b201708220952_create_project_label
INFO - [doRecordOnly] b201709121141_delete_project_classification_widget
==============================================
Installation completed successfully!
- WHAT TO DO TO FINISH THE TULEAP INSTALLATION (see /root/todo_tuleap.txt)
- Your domain name is localhost
- Automatically generated passwords are stored in /data/root/.tuleap_passwd
- Customize /etc/tuleap/conf/local.inc and /etc/tuleap/conf/database.inc
- You may also want to customize /etc/httpd/conf/httpd.conf
- Customize /etc/tuleap/site-content information for your site.
-   For instance: contact/contact.txt 
- 
- Default admin credentials are login: admin / password: nGgYkQiMuLzptZe
- CHANGE DEFAULT CREDENTIALS BEFORE FIRST USAGE
- Auto generated passwords (mysql, application, etc) are stored in /data/root/.tuleap_passwd
- 
- -----------------------------------------
- This TODO list is available in /root/todo_tuleap.txt.
Generating SSH2 RSA host key:                              [  OK  ]
Generating SSH1 RSA host key:                              [  OK  ]
Generating SSH2 DSA host key:                              [  OK  ]
Starting sshd:                                             [  OK  ]
Stopping sshd:                                             [  OK  ]
Stopping mysqld:                                           [  OK  ]
Stopping httpd:                                            [  OK  ]
Stopping crond:                                            [  OK  ]
mv: cannot remove `/etc/httpd/conf': Directory not empty
+ ./boot-fixpath.sh
rm: cannot remove `/etc/httpd/conf': Directory not empty
+ ./boot-update-config.sh
+ perl -pi -e 's%^short_open_tag = Off%short_open_tag = On%' /etc/php.ini
+ perl -pi -e 's%^;date.timezone =%date.timezone = Europe/Paris%' /etc/php.ini
+ perl -pi -e 's%^#myhostname = host.domain.tld%myhostname = localhost%' /etc/postfix/main.cf
+ perl -pi -e 's%^alias_maps = hash:/etc/aliases%alias_maps = hash:/etc/aliases,hash:/etc/aliases.codendi%' /etc/postfix/main.cf
+ perl -pi -e 's%^alias_database = hash:/etc/aliases%alias_database = hash:/etc/aliases,hash:/etc/aliases.codendi%' /etc/postfix/main.cf
+ perl -pi -e 's%^#recipient_delimiter = %recipient_delimiter = %' /etc/postfix/main.cf
+ perl -pi -e 's%enable-cache[\t ]+group[\t ]+yes%enable-cache group no%' /etc/nscd.conf
+ source mysql-utils.sh
++ set -e
+ start_mysql
+ echo 'Start mysql'
Start mysql
+ sleep 1
+ /usr/bin/python /usr/lib/python2.6/site-packages/supervisor/pidproxy.py /var/run/mysqld/mysqld.pid /usr/bin/mysqld_safe
170919 09:32:38 mysqld_safe Logging to '/var/lib/mysql/b82d1a88bcb7.err'.
chown: invalid user: `mysql'
170919 09:32:38 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
170919 09:32:38 mysqld_safe mysqld from pid file /var/lib/mysql/b82d1a88bcb7.pid ended
++ egrep '^\$sys_dbpasswd' /etc/tuleap/conf/database.inc
++ sed -e 's/^\$sys_dbpasswd="\(.*\)";$/\1/'
egrep: /etc/tuleap/conf/database.inc: No such file or directory
+ mysql -ucodendiadm -p -e 'show databases'
Enter password: 
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)
+ echo 'Wait for the db'
Wait for the db
+ sleep 1
++ egrep '^\$sys_dbpasswd' /etc/tuleap/conf/database.inc
++ sed -e 's/^\$sys_dbpasswd="\(.*\)";$/\1/'
egrep: /etc/tuleap/conf/database.inc: No such file or directory
+ mysql -ucodendiadm -p -e 'show databases'
Enter password: 
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)
+ echo 'Wait for the db'
Wait for the db
+ sleep 1
++ egrep '^\$sys_dbpasswd' /etc/tuleap/conf/database.inc
++ sed -e 's/^\$sys_dbpasswd="\(.*\)";$/\1/'
egrep: /etc/tuleap/conf/database.inc: No such file or directory
+ mysql -ucodendiadm -p -e 'show databases'
Enter password: 
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)
+ echo 'Wait for the db'
Wait for the db
+ sleep 1
++ egrep '^\$sys_dbpasswd' /etc/tuleap/conf/database.inc
++ sed -e 's/^\$sys_dbpasswd="\(.*\)";$/\1/'
egrep: /etc/tuleap/conf/database.inc: No such file or directory
+ mysql -ucodendiadm -p -e 'show databases'
Enter password: 
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)
+ echo 'Wait for the db'
Wait for the db
+ sleep 1
++ egrep '^\$sys_dbpasswd' /etc/tuleap/conf/database.inc
++ sed -e 's/^\$sys_dbpasswd="\(.*\)";$/\1/'
egrep: /etc/tuleap/conf/database.inc: No such file or directory
+ mysql -ucodendiadm -p -e 'show databases'
Enter password:
```

**Comment by tcottier-enalean** [comment]
Hello,

Try to remove and recreate your tuleap-data

**Comment by tcottier-enalean** [comment]
Sorry wrong manipulation. I'll have a look at your problem.

**Comment by tcottier-enalean** [comment]
The real problem is:

```
mv: inter-device move failed: `/var/lib/mysql' to `/data/lib/mysql'; unable to remove target: Directory not empty
mv: inter-device move failed: `/var/lib/gitolite' to `/data/lib/gitolite'; unable to remove target: Directory not empty
```

Is tuleap-data a new volume or does it contain previous data?
If this is your first installation, try to remove this volume and start the process again.

**Comment by RealSnowbird** [comment]
Thanks for commenting. I've tried to remove and recreate the volume but the result was the same.
Any other idea what I could check?

**Comment by tcottier-enalean** [comment]
Do you still have the following line on your logs?
```
mv: inter-device move failed: `/var/lib/mysql' to `/data/lib/mysql'; unable to remove target: Directory not empty
```

**Comment by RealSnowbird** [comment]
Sorry for being unclear. I did not get this error again but several other error messages. In the end, I'm asked to enter a password but the password provided in the log doesn't work. The file /data/root/.tuleap_passwd doesn't exist (the folders /data/root do exist).

Before, I've deleted volume, container and image and then executed

docker volume create --name tuleap-data
docker run -ti -e VIRTUAL_HOST=localhost -P -v tuleap-data:/data enalean/tuleap-aio

This is the log:

ERRO[498402] Handler for POST /v1.27/containers/create returned error: No such image: enalean/tuleap-aio:latest 
Unable to find image 'enalean/tuleap-aio:latest' locally
latest: Pulling from enalean/tuleap-aio
b26de5a391ad: Pull complete 
21dabc0355f9: Pull complete 
79caa8c91849: Pull complete 
13b40ecf9ecc: Pull complete 
dbc92eb11c79: Pull complete 
715f5c83bdd1: Pull complete 
Digest: sha256:22ac2beb22244f1df7a15e91bbcc09064a8ae79db5d00327093099324b14725b
Status: Downloaded newer image for enalean/tuleap-aio:latest
+ TULEAP_INSTALL_TIME=false
+ '[' '!' -f /data/etc/tuleap/conf/local.inc ']'
+ TULEAP_INSTALL_TIME=true
+ ./boot-install.sh
mv: cannot remove `/var/lib/gitolite/.gitolite/conf': Directory not empty
mv: cannot remove `/var/lib/gitolite/.gitolite/hooks/common': Directory not empty
mv: cannot remove `/var/lib/gitolite/.gitolite/hooks/gitolite-admin': Directory not empty
mv: cannot remove `/var/lib/gitolite/.gitolite/keydir': Directory not empty
mv: cannot remove `/var/lib/gitolite/.gitolite/logs': Directory not empty
/bin/ls: cannot access /root/app/patches/*.patch: No such file or directory
Good! You are running CentOS 6.9

Configuration questions

Changing password for user codendiadm.
passwd: all authentication tokens updated successfully.
Installing Tuleap configuration files...
Renaming existing Apache configuration files...
Installing Apache configuration files...
Installing log files rotation...
Installing NSS configuration files...
Creating MySQL conf file...
Initializing MySQL: You can ignore additionnal messages on MySQL below this line:
***************************************
Stopping mysqld:                                           [  OK  ]
Initializing MySQL database:  Installing MySQL system tables...
/usr/libexec/mysqld: Got an error reading communication packets
OK
Filling help tables...
/usr/libexec/mysqld: Got an error reading communication packets
OK

To start mysqld at boot time you have to copy
support-files/mysql.server to the right place for your system

PLEASE REMEMBER TO SET A PASSWORD FOR THE MySQL root USER !
To do so, start the server, then issue the following commands:

/usr/bin/mysqladmin -u root password 'new-password'
/usr/bin/mysqladmin -u root -h 6a4b3dcb6e10 password 'new-password'

Alternatively you can run:
/usr/bin/mysql_secure_installation

which will also give you the option of removing the test
databases and anonymous user created by default.  This is
strongly recommended for production servers.

See the manual for more instructions.

You can start the MySQL daemon with:
cd /usr ; /usr/bin/mysqld_safe &

You can test the MySQL daemon with mysql-test-run.pl
cd /usr/mysql-test ; perl mysql-test-run.pl

Please report any problems with the /usr/bin/mysqlbug script!

                                                           [  OK  ]
Starting mysqld:                                           [  OK  ]
***************************************
Creating the Tuleap database...
Populating the Tuleap database...
PHP Fatal error:  Call to undefined function random_bytes() in /usr/share/tuleap/src/common/user/RandomNumberGenerator.class.php on line 38
Configuring the CVS server and CVS tracking tools...
/bin/cp: cannot stat `/etc/xinetd.d/cvs': No such file or directory
Stopping xinetd:                                           [FAILED]
Starting xinetd: /etc/init.d/xinetd: line 66: warning: setlocale: LC_TIME: cannot change locale (en_US): No such file or directory
/etc/init.d/xinetd: line 67: warning: setlocale: LC_ALL: cannot change locale (en_US): No such file or directory
/etc/init.d/xinetd: line 72: warning: setlocale: LC_ALL: cannot change locale (en_US): No such file or directory
/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US): No such file or directory
                                                           [  OK  ]
Updating daily cron job in system crontab...
Use same-host project web sites
Restore application cache directories:
info Recreating /var/tmp/tuleap_cache/lang
info Restore ownership to /var/tmp/tuleap_cache
info Restore ownership to /var/tmp/tuleap_cache/lang
info Restore ownership to /var/tmp/tuleap_cache/hooks.json
Invalidate application caches:
info Invalidate template engine cache
info Invalidate Restler cache
info Invalidate language cache
info Invalidate WSDL
info Invalidate Plugin hooks
Starting Tuleap: [OK]
Stopping httpd:                                            [  OK  ]
Starting httpd: Syntax error on line 65 of /etc/httpd/conf.d/tuleap-vhost.conf:
SSLCertificateFile: file '/etc/pki/tls/certs/localhost.crt' does not exist or is empty
                                                           [FAILED]
Starting crond:                                            [  OK  ]
Stopping nscd:                                             [FAILED]
Starting nscd:                                             [  OK  ]
nscd: 703 monitoring file `/etc/passwd` (1)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/group` (3)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/hosts` (4)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/resolv.conf` (5)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 monitoring file `/etc/services` (6)
nscd: 703 monitoring directory `/etc` (2)
nscd: 703 disabled inotify-based monitoring for file `/etc/netgroup': No such file or directory
nscd: 703 disabled monitoring for file `/etc/netgroup': No such file or directory
Install tuleap plugins
Install Docman
Install tracker
Import succeeded
Import succeeded
Import succeeded
error: The short name 'patch' is already used. Please use another one.
Import succeeded
Import succeeded
Import succeeded
Import succeeded
Import succeeded
Install Graphontrackersv5
Install AgileDashboard
Install Git
Creating Scrum template
INFO - [doRecordOnly] b201007220743_add_table_notification_delegation
INFO - [doRecordOnly] b201007291109_modify_pk_table_notification_delegation
INFO - [doRecordOnly] b201008200802_reduce_load_on_user_table
INFO - [doRecordOnly] b201009280731_add_table_groups_notif_delegation_message
INFO - [doRecordOnly] b201010191436_add_table_frs_file_deleted
INFO - [doRecordOnly] b201010201546_fill_frs_file_deleted
INFO - [doRecordOnly] b201011230835_add_column_format_to_artifact_history
INFO - [doRecordOnly] b201012140821_improve_frs_file
INFO - [doRecordOnly] b201012240808_add_table_frs_log
INFO - [doRecordOnly] b201103081738_add_column_filepath_to_frs_file
INFO - [doRecordOnly] b201103281122_add_wiki_attachment_delete_mechanism
INFO - [doRecordOnly] b201108241423_remove_slmbug_and_story
INFO - [doRecordOnly] b201108311456_add_image_renderer
INFO - [doRecordOnly] b201111021759_id_sharing
INFO - [doRecordOnly] b201201181909_add_index_on_svn_commits_table
INFO - [doRecordOnly] b201201231552_add_table_soap_limits
INFO - [doRecordOnly] b201203140933_add_notif_svn_per_branch
INFO - [doRecordOnly] b201207051342_add_has_avatar_column
INFO - [doRecordOnly] b201208241130_add_anonymous_in_filemodule_monitor
INFO - [doRecordOnly] b201209131359_add_ugroup_binding_column
INFO - [doRecordOnly] b201210121809_purge_bad_sshkeys_from_db
INFO - [doRecordOnly] b201302091402_add_frs_file_comment
INFO - [doRecordOnly] b201303121639_add_owner_to_system_events
INFO - [doRecordOnly] b201305071724_add_svn_change_log_option
INFO - [doRecordOnly] b201305161743_remove_unique_from_svn_checkins
INFO - [doRecordOnly] b201306110951_add_parent_child_project_link_table
INFO - [doRecordOnly] b201306171145_add_generic_user_table
INFO - [doRecordOnly] b201310171433_add_rest_authentication_token_table
INFO - [doRecordOnly] b201311081011_remove_codex_themes
INFO - [doRecordOnly] b201403061540_add_forge_permissions_table
INFO - [doRecordOnly] b201404281427_rename_experimental_to_flamingparrot
INFO - [doRecordOnly] b201405051118_add_svn_accessfile_history_table_with_first_version
INFO - [doRecordOnly] b201405051133_modify_groups_table_to_use_svn_accessfile_history_id
INFO - [doRecordOnly] b201405061202_modify_svn_accessfile_history_table
INFO - [doRecordOnly] b201405071410_replace_svn_accessfile_column_name_in_groups
INFO - [doRecordOnly] b201405071644_replace_version_date_column_values_by_timestamp
INFO - [doRecordOnly] b201405141602_remove_fontsize_preference
INFO - [doRecordOnly] b201406041516_add_email_gateway_salt
INFO - [doRecordOnly] b201407040902_old_user_do_not_see_welcome_tour
INFO - [doRecordOnly] b201407041153_add_tour_usage_statistic_table
INFO - [doRecordOnly] b201407081125_add_missing_column_in_tour_usage_statistics
INFO - [doRecordOnly] b201410301650_add_commit_to_tag_denied_in_groups
INFO - [doRecordOnly] b201503061743_add_homepage
INFO - [doRecordOnly] b201503061743_add_homepage_headline
INFO - [doRecordOnly] b201503271456_add_forgeconfig
INFO - [doRecordOnly] b201504081720_allow_restricted_project_access
INFO - [doRecordOnly] b201504091644_add_authenticated_user_group
INFO - [doRecordOnly] b201504171343_restore_FRS_owner_group
INFO - [doRecordOnly] b201505051630_add_new_password_field
INFO - [doRecordOnly] b201505131103_project_admin_can_choose_visibility
INFO - [doRecordOnly] b201505302300_remove_twitter_widgets
INFO - [doRecordOnly] b201506011450_increase_unix_password_field_size
INFO - [doRecordOnly] b201506031610_add_svn_immutable_tags_whitelist_table
INFO - [doRecordOnly] b201506031658_enlarge_your_tinyint
INFO - [doRecordOnly] b201507152200_increase_session_ip_field_size
INFO - [doRecordOnly] b201508141600_remove_people_skills
INFO - [doRecordOnly] b201508171048_add_svn_immutable_tags_table
INFO - [doRecordOnly] b201508241048_add_truncated_emails_column
INFO - [doRecordOnly] b201508271443_add_index_on_system_event_table
INFO - [doRecordOnly] b201509020939_drop_priority_plugin_hook
INFO - [doRecordOnly] b201510011056_add_index_on_service_table
INFO - [doRecordOnly] b201510131542_turn_user_groups_tables_innodb
INFO - [doRecordOnly] b201511231728_add_trovecat_mandatory_column
INFO - [doRecordOnly] b201512171700_add_svn_token_table
INFO - [doRecordOnly] b201512211530_force_refresh_svnaccessfile
INFO - [doRecordOnly] b201512281119_add_svn_token_usage_table
INFO - [doRecordOnly] b201512291430_add_indexes_for_statistics
INFO - [doRecordOnly] b201601051638_add_trovecat_display_column
INFO - [doRecordOnly] b201607221530_cleanup_csrf_token_from_userprefs
INFO - [doRecordOnly] b201607281135_create_plugin_frs_global_permissions
INFO - [doRecordOnly] b201607281200_migrate_existing_permissions
INFO - [doRecordOnly] b201608041530_migrate_existing_read_permissions
INFO - [doRecordOnly] b201608091610_add_index_for_frs_permissions
INFO - [doRecordOnly] b201608291800_migrate_incoherent_permissions_for_release
INFO - [doRecordOnly] b201608298050_migrate_incoherent_permissions
INFO - [doRecordOnly] b201609082200_remove_admssw_plugin
INFO - [doRecordOnly] b201609082230_remove_boomerang_plugin
INFO - [doRecordOnly] b201609082230_remove_foafprofiles_plugin
INFO - [doRecordOnly] b201609131045_update_session_management
INFO - [doRecordOnly] b201609221505_add_missing_permissions_french_wiki_pages
INFO - [doRecordOnly] b201610121840_remove_fusionforgecompat_plugin
INFO - [doRecordOnly] b201610241145_update_frs_admin_description
INFO - [doRecordOnly] b201611031112_add_svn_cache_parameter_table
INFO - [doRecordOnly] b201612231324_update_trove_categories_incorrect_values
INFO - [doRecordOnly] b201701131030_create_table_user_lost_password
INFO - [doRecordOnly] b201701231415_update_public_forum_in_private_projects
INFO - [doRecordOnly] b201701251415_user_lost_password_add_creation_date
INFO - [doRecordOnly] b201701311615_feedback_supports_new_session_identifier
INFO - [doRecordOnly] b20170207_remove_snippet_reference
INFO - [doRecordOnly] b201703131130_create_table_project_webhook_url
INFO - [doRecordOnly] b201703141400_create_table_project_webhook_log
INFO - [doRecordOnly] b201703201030_add_project_webhook_name
INFO - [doRecordOnly] b201703281110_keep_only_modperl_svn_authentication_mode
INFO - [doRecordOnly] b201704041400_clean_recent_element_user_preferences
INFO - [doRecordOnly] b201704051330_remove_binary_serialized_feedback
INFO - [doRecordOnly] b201705031528_create_table_user_dashboards
INFO - [doRecordOnly] b201705110903_create_table_project_dashboards
INFO - [doRecordOnly] b201705121210_update_table_user_dashboards
INFO - [doRecordOnly] b201705121719_update_table_project_dashboards
INFO - [doRecordOnly] b201705161407_create_tables_dashboards_lines
INFO - [doRecordOnly] b201705161407_create_tables_dashboards_lines_columns
INFO - [doRecordOnly] b201705161407_create_tables_dashboards_lines_columns_widgets
INFO - [doRecordOnly] b201705181212_store_widget_name
INFO - [doRecordOnly] b201705231141_store_widget_is_minimized
INFO - [doRecordOnly] b201706091011_migrate_old_dashboard
INFO - [doRecordOnly] b201706301720_remove_docman_v1
INFO - [doRecordOnly] b201708180900_remove_invalid_china_timezones
INFO - [doRecordOnly] b201708220952_create_project_label
INFO - [doRecordOnly] b201709121141_delete_project_classification_widget
==============================================
Installation completed successfully!
- WHAT TO DO TO FINISH THE TULEAP INSTALLATION (see /root/todo_tuleap.txt)
- Your domain name is localhost
- Automatically generated passwords are stored in /data/root/.tuleap_passwd
- Customize /etc/tuleap/conf/local.inc and /etc/tuleap/conf/database.inc
- You may also want to customize /etc/httpd/conf/httpd.conf
- Customize /etc/tuleap/site-content information for your site.
-   For instance: contact/contact.txt 
- 
- Default admin credentials are login: admin / password: 0p5tSBdVYNorn43
- CHANGE DEFAULT CREDENTIALS BEFORE FIRST USAGE
- Auto generated passwords (mysql, application, etc) are stored in /data/root/.tuleap_passwd
- 
- -----------------------------------------
- This TODO list is available in /root/todo_tuleap.txt.
Generating SSH2 RSA host key:                              [  OK  ]
Generating SSH1 RSA host key:                              [  OK  ]
Generating SSH2 DSA host key:                              [  OK  ]
Starting sshd:                                             [  OK  ]
Stopping sshd:                                             [  OK  ]
./boot-install.sh: line 63: /var/lib/gitolite/.ssh/authorized_keys: No such file or directory
+ ./boot-fixpath.sh
rm: cannot remove `/etc/httpd/conf': Directory not empty
+ ./boot-update-config.sh
+ perl -pi -e 's%^short_open_tag = Off%short_open_tag = On%' /etc/php.ini
+ perl -pi -e 's%^;date.timezone =%date.timezone = Europe/Paris%' /etc/php.ini
+ perl -pi -e 's%^#myhostname = host.domain.tld%myhostname = localhost%' /etc/postfix/main.cf
+ perl -pi -e 's%^alias_maps = hash:/etc/aliases%alias_maps = hash:/etc/aliases,hash:/etc/aliases.codendi%' /etc/postfix/main.cf
+ perl -pi -e 's%^alias_database = hash:/etc/aliases%alias_database = hash:/etc/aliases,hash:/etc/aliases.codendi%' /etc/postfix/main.cf
+ perl -pi -e 's%^#recipient_delimiter = %recipient_delimiter = %' /etc/postfix/main.cf
+ perl -pi -e 's%enable-cache[\t ]+group[\t ]+yes%enable-cache group no%' /etc/nscd.conf
+ source mysql-utils.sh
++ set -e
+ start_mysql
+ echo 'Start mysql'
Start mysql
+ sleep 1
+ /usr/bin/python /usr/lib/python2.6/site-packages/supervisor/pidproxy.py /var/run/mysqld/mysqld.pid /usr/bin/mysqld_safe
170927 09:35:04 mysqld_safe Logging to '/var/lib/mysql/6a4b3dcb6e10.err'.
chown: invalid user: `mysql'
170927 09:35:04 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
170927 09:35:04 mysqld_safe mysqld from pid file /var/lib/mysql/6a4b3dcb6e10.pid ended
++ egrep '^\$sys_dbpasswd' /etc/tuleap/conf/database.inc
++ sed -e 's/^\$sys_dbpasswd="\(.*\)";$/\1/'
egrep: /etc/tuleap/conf/database.inc: No such file or directory
+ mysql -ucodendiadm -p -e 'show databases'
Enter password:

**Comment by superbiche** [comment]
Hi, I'm having the exactly same issue (first one, not the new one), same thing after deleting volume/containers and recreating.

Docker CE Version 17.09.0-ce-mac35 (19611) on Mac OS Sierra 10.12.6

**Comment by bargacha** [comment]
Hi ! Same problem for me. Did you find any solution for this issue ?

Docker CE 17.09.0-ce build afdb6d4 on Ubuntu 16.04

**Comment by vaceletm** [comment]
I think I found the source of the issue.

A build is on going for `latest` tag: https://hub.docker.com/r/enalean/tuleap-aio/builds/

Once it's done, could you try again ?

**Comment by bargacha** [comment]
Of course !
Keep you in touch

**Comment by bargacha** [comment]
Everything works fine now with the last build.
Thank you for your reactivity !

**Comment by vaceletm** [comment]
Thanks for the tests

**Comment by sreecodeslayer** [comment]
Am not sure if this is fixed, just test on Docker / Ubuntu 16.04 on Amazon EC2, as I write, and I get all those errors mention in the OP:

Command used is as same as in the README: `docker run -ti -e VIRTUAL_HOST=localhost -p 80:80 -p 443:443 -p 22:22 -v tuleap-data:/data enalean/tuleap-aio`
Now, EC2 has default SSH port opened as 22, and will be in use, so I used 33 (Which I opened) the next time,
`docker run -ti -e VIRTUAL_HOST=localhost -p 80:80 -p 443:443 -p 33:33 -v tuleap-data:/data enalean/tuleap-aio`

I do not wish to flood the thread with lengthy Log, but here is some of the errors am seeing.

```
.
.
Good! You are running CentOS 6.9
.
.
Installation completed successfully!
- WHAT TO DO TO FINISH THE TULEAP INSTALLATION (see /root/todo_tuleap.txt)
- Your domain name is localhost
- Automatically generated passwords are stored in /data/root/.tuleap_passwd
- Customize /etc/tuleap/conf/local.inc and /etc/tuleap/conf/database.inc
- You may also want to customize /etc/httpd/conf/httpd.conf
- Customize /etc/tuleap/site-content information for your site.
-   For instance: contact/contact.txt
-
- Default admin credentials are login: admin / password: ZHStbj8jU0QyoiN
- CHANGE DEFAULT CREDENTIALS BEFORE FIRST USAGE
- Auto generated passwords (mysql, application, etc) are stored in /data/root/.tuleap_passwd
-
- -----------------------------------------
- This TODO list is available in /root/todo_tuleap.txt.
Generating SSH2 RSA host key:                              [  OK  ]
Generating SSH1 RSA host key:                              [  OK  ]
Generating SSH2 DSA host key:                              [  OK  ]
Starting sshd:                                             [  OK  ]
Stopping sshd:                                             [  OK  ]
Stopping mysqld:                                           [  OK  ]
Stopping httpd:                                            [  OK  ]
Stopping crond:                                            [  OK  ]
Stopping nginx:                                            [  OK  ]
Stopping php-fpm:                                          [  OK  ]
Update data ownership to current image...chown: cannot access `/data/etc/httpd/conf.d/codendi_svnroot.conf': No such file or directory
chown: cannot access `/data/etc/httpd/conf.d/codendi_svnroot.conf.old': No such file or directory
chown: cannot access `/data/lib/tuleap/boomerang': No such file or directory
chown: cannot access `/data/lib/tuleap/forumml': No such file or directory
chown: cannot access `/data/lib/tuleap/trackerv3': No such file or directory
chown: cannot access `/data/lib/tuleap/secure_ftp/*': No such file or directory
chown: cannot access `/data/lib/tuleap/svnroot/*': No such file or directory
DONE !
Restore application cache directories:
PHP Warning:  mysql_connect(): No such file or directory in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php on line 112

Warning: mysql_connect(): No such file or directory in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php on line 112
PHP Warning:  mysql_error() expects parameter 1 to be resource, boolean given in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php on line 91

Warning: mysql_error() expects parameter 1 to be resource, boolean given in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php on line 91
PHP Fatal error:  Uncaught exception 'DataAccessException' with message 'Unable to access the database ( - 2002). Please contact your administrator.' in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php:91
Stack trace:
#0 /usr/share/tuleap/src/common/dao/include/DataAccess.class.php(72): DataAccess->reconnect()
#1 /usr/share/tuleap/src/common/dao/CodendiDataAccess.class.php(41): DataAccess->DataAccess(Object(DataAccessCredentials), 0)
#2 /usr/share/tuleap/src/common/dao/CodendiDataAccess.class.php(49): CodendiDataAccess->__construct()
#3 /usr/share/tuleap/src/common/dao/include/DataAccessObject.class.php(50): CodendiDataAccess::instance()
#4 /usr/share/tuleap/src/common/Config/ForgeConfig.php(53): DataAccessObject->__construct()
#5 /usr/share/tuleap/src/www/include/pre.php(48): ForgeConfig::loadFromDatabase()
#6 /usr/share/tuleap/src/utils/tuleap.php(61): require_once('/usr/share/tule...')
#7 {main}
  thrown in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php on line 91

Fatal error: Uncaught exception 'DataAccessException' with message 'Unable to access the database ( - 2002). Please contact your administrator.' in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php:91
Stack trace:
#0 /usr/share/tuleap/src/common/dao/include/DataAccess.class.php(72): DataAccess->reconnect()
#1 /usr/share/tuleap/src/common/dao/CodendiDataAccess.class.php(41): DataAccess->DataAccess(Object(DataAccessCredentials), 0)
#2 /usr/share/tuleap/src/common/dao/CodendiDataAccess.class.php(49): CodendiDataAccess->__construct()
#3 /usr/share/tuleap/src/common/dao/include/DataAccessObject.class.php(50): CodendiDataAccess::instance()
#4 /usr/share/tuleap/src/common/Config/ForgeConfig.php(53): DataAccessObject->__construct()
#5 /usr/share/tuleap/src/www/include/pre.php(48): ForgeConfig::loadFromDatabase()
#6 /usr/share/tuleap/src/utils/tuleap.php(61): require_once('/usr/share/tule...')
#7 {main}
  thrown in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php on line 91
.
.
Fatal error: Uncaught exception 'DataAccessException' with message 'Unable to access the database ( - 2002). Please contact your administrator.' in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php:91
Stack trace:
#0 /usr/share/tuleap/src/common/dao/include/DataAccess.class.php(72): DataAccess->reconnect()
#1 /usr/share/tuleap/src/common/dao/CodendiDataAccess.class.php(41): DataAccess->DataAccess(Object(DataAccessCredentials), 0)
#2 /usr/share/tuleap/src/common/dao/CodendiDataAccess.class.php(49): CodendiDataAccess->__construct()
#3 /usr/share/tuleap/src/common/dao/include/DataAccessObject.class.php(50): CodendiDataAccess::instance()
#4 /usr/share/tuleap/src/common/Config/ForgeConfig.php(53): DataAccessObject->__construct()
#5 /usr/share/tuleap/src/www/include/pre.php(48): ForgeConfig::loadFromDatabase()
#6 /usr/share/tuleap/src/utils/tuleap.php(61): require_once('/usr/share/tule...')
#7 {main}
  thrown in /usr/share/tuleap/src/common/dao/include/DataAccess.class.php on line 91
Starting Tuleap: [OK]
/
2017-11-21 11:29:45,342 CRIT Supervisor running as root (no user in config file)
2017-11-21 11:29:45,355 INFO supervisord started with pid 1
2017-11-21 11:29:45,356 INFO spawned: 'httpd' with pid 1331
2017-11-21 11:29:45,358 INFO spawned: 'sshd' with pid 1332
2017-11-21 11:29:45,359 INFO spawned: 'postfix' with pid 1333
2017-11-21 11:29:45,360 INFO spawned: 'fpm' with pid 1334
2017-11-21 11:29:45,370 INFO spawned: 'nginx' with pid 1335
2017-11-21 11:29:45,372 INFO spawned: 'rsyslog' with pid 1336
2017-11-21 11:29:45,386 INFO spawned: 'mysql' with pid 1337
2017-11-21 11:29:45,387 INFO spawned: 'crond' with pid 1338
2017-11-21 11:29:45,405 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2017-11-21 11:29:46,551 INFO success: httpd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-11-21 11:29:46,552 INFO success: sshd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-11-21 11:29:46,552 INFO success: fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-11-21 11:29:46,552 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-11-21 11:29:46,552 INFO success: mysql entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-11-21 11:29:46,553 INFO success: crond entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-11-21 11:29:47,656 INFO exited: postfix (exit status 0; expected)
2017-11-21 11:29:47,656 INFO received SIGCLD indicating a child quit
2017-11-21 11:29:50,660 INFO success: rsyslog entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)

```
Let me know if there is more info I can provide. :smile:

**Comment by vaceletm** [comment]
It's not the same issue, the one you report is described here https://tuleap.net/plugins/tracker/?aid=10824 and is currently in review. 

While the errors are not nice, they have no impact on the platform ATM. They will disappear once the fix is merged + next reboot.

I'm gonna lock this thread to avoid a catch all of "errors at start". If you find a new issue please create a new entry.
